### ISSUE TITLE
Accessibility: 

### ISSUE THREAD

**Issue by DBAKITA** [jupyterhub/outreachy#130]
### What page is this for?

https://ipython.org/documentation.html

### WAVE accessibility report

![wave evaluation of ipython documentation](https://user-images.githubusercontent.com/87822858/199071254-1e4e64ed-fa31-4c43-a0b6-a923b8d4d1cb.JPG)

I evaluated https://ipython.org/documentation.html using WAVE web accessibility evaluation tool.

Link to my  comment https://github.com/jupyterhub/outreachy/issues/38#issuecomment-1277649881

Below are the findings and the suggested changes to improve the page's accessibility.

**1. Language Missing or Invalid** (The attribute is missing or is empty, or a lang attribute value is not a valid language identifier.)

Suggestion: The document language should be identified using the attribute with a vald value (e.g., ). Identifying the language of the page or page elements will allow screen readers to read the content in the appropriate language. It also facilitates automatic translation of content.

**2. Contrast Errors** (Fifty 54 of them) - Very low contrast between text and background colors.

Suggestion: Providing adequate contrast of text for all users, especially users with low vision. The contrast between the foreground (text) color and the background color should be increased. In regards to text size, large text (larger than 18 point or 14 point bold) does not require as much contrast as smaller text.

**3.There is justified text in 6 places.**

Suggestion: Remove the full justification from the text. This is because large blocks of justified text can negatively impact readability due to varying word/letter spacing and 'rivers of white' that flow through the text.

**4. One suspicious alternative text**

Suggestion: If the alternative text for an image does not provide the same content or information conveyed by the image, that content will not be available to screen reader users and when images are unavailable. The alternative text for the image or image input should provide an appropriate and equivalent alternative to the content and function of the image. Screen readers and browser presentation inform the user that the object is an image, so alternative text of "image of..." (and similar) should be avoided. If the image does not convey content or if the content is presented in nearby text (e.g., a caption), null/empty alternative text (alt="") will be appropriate.

**5. Images having a title attribute value but no alt value.**

Suggestion: Ensure the title attribute value presents the content and function of images. For improved accessibility, the alt attribute should be used when possible.

**6. The web page contains Links to PDF documents.**- PDF documents often have accessibility issues and typically require using a separate application or plug-in to view them. This can cause confusion and navigation difficulties.

Suggestion:

Ensuring the PDF document is natively accessible.

Also informing users that the link will open a PDF document.

-HTML content should often be used in place of or in addition to the PDF document.

**7. Alternative text is present for 2 images images that are within link.**

Suggestion: including appropriate alternative text on an image within a link to ensure that the function and purpose of the link and the content of the image is available to screen reader users or when images are unavailable.

**8. Some headings are not structured well in the page**

Suggestion: This can be improved by ensuring that where headings are used, the text in question is actually a heading and that it is structured correctly in the page outline. When this is put into consideration, the headings will facilitate page navigation for users of assistive technologies. The headings will also provide semantic and visual meaning and structure to the document.

**9. Un Ordered lists were used**

Suggestion: Where list items are sequential or numbered, an ordered list is likely more appropriate.
We should use unordered (bulleted) lists if they are appropriate for the context.

**10. Little information is presented on the website footer.**

Suggestion: Including more information in the footer will provide site visitors with a sense of consistency, as the same information will appear at the bottom of every single page. This helps people know that they’re really on the web site and makes it easy for repeat visitors to find what they’re looking for. I suggest that some calls to action be added to the footer for example subscribe to our newsletter, learn more, get in touch, leave a review, Share this post and others, links to social accounts among others.
-Considering color contrast while adding information in the footer. Making sure text and images are different enough from your footer background that they can be easily consumed. Avoiding dark text on a dark background or light text on a light background. Using more than just color to differentiate links from other text. (this can be done with underlines, icons, and other cues) Including alt text for any images or graphics in the footer. Screen readers with visual impairments will use these to “read” images
-Also by Sticking to simple, easy-to-read fonts that everyone can consume while making sure they’re large enough so that anyone with vision difficulties can still read every word.

**11. An Aria Lable was used.**

Suggestion: I suggest that where possible, the standard HTML label should be used.

### The HTML element to be changed

```html
1.Current html of the page with the language attribute missing 
<html xmlns="http://www.w3.org/1999/xhtml">
<head>

2. Current html of Images having a title attribute value but no alt value.

<a href="books.html"><img
title="Packt Publishing donates a portion of the proceeds from this book to
support IPython's development." width="160" height="200" id="sidebar-book-cover"
style="margin-bottom: 10px;"></a>

3.Current html of PDF documents on the page (under documentation for older releases)

The current html of the pdf links under documentation of older releases
<li>0.11: <a class="reference external" href="https://ipython.org/ipython-doc/rel-0.11/index.html">HTML</a> and <a class="reference external" href="https://ipython.org/ipython-doc/rel-0.11/ipython.pdf">PDF</a>.
</li>

<li>0.10.2: <a class="reference external" href="https://ipython.org/ipython-doc/rel-0.10.2/html">HTML</a> and <a class="reference external" href="https://ipython.org/ipython-doc/rel-0.10.2/ipython.pdf">PDF</a>.</li>

<li>0.10.1: <a class="reference external" href="https://ipython.org/ipython-doc/rel-0.10.1/html">HTML</a> and <a class="reference external" href="https://ipython.org/ipython-doc/rel-0.10.1/ipython.pdf">PDF</a>.</li>

<li>0.10: <a class="reference external" href="https://ipython.org/ipython-doc/rel-0.10/html">HTML</a> and <a class="reference external" href="https://ipython.org/ipython-doc/rel-0.10/ipython.pdf">PDF</a>.</li>

<li>0.9.1: <a class="reference external" href="https://ipython.org/ipython-doc/rel-0.9.1/html">HTML</a> and <a class="reference external" href="https://ipython.org/ipython-doc/rel-0.9.1/ipython.pdf">PDF</a>.</li>

<li>0.9: <a class="reference external" href="https://ipython.org/ipython-doc/rel-0.9/html">HTML</a> and <a class="reference external" href="https://ipython.org/ipython-doc/rel-0.9/ipython.pdf">PDF</a>.</li>

4. . Alternative text is present for 2 images that are within link.
Current html of first image
<img class="logo" src="_static/IPy_header.png" alt="Logo"/>

Current html of second image
<img src="_static/jupytercon-logo.svg" width="180px" alt="JupyterCon 2017">

5.  Little information is presented on the website footer.
Current css of the footer

/* Footer */

div.footer-wrapper {
  padding-top: 10px;
  padding-bottom: 10px;
}

div.footer {
  border-top: 2px solid #aaa;
  text-align: right;
}

div.footer, div.footer a {
  color: #888a85;
}

Current Html of the footer 
<div class="footer-wrapper">
    
  <div class="footer" role="contentinfo">
      &#169; Copyright the IPython development team.
    Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.8.5.
  </div>
  </div>

HTML of the current footer
<div class="footer-wrapper">
    
  <div class="footer" role="contentinfo">
      &#169; Copyright the IPython development team.
    Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.8.5.
  </div>
  </div>
6. An Aria Lable was used where it should not be
current html 
<div class="sphinxsidebar" role="navigation" aria-label="main navigation">

7. there is no attribute "data-url_root"
current html 
<script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>

8.required attribute "type" not specified

Current html 
<script type src="_static/nbviewer_thm.js"></script>
```

### Your proposed HTML (or CSS), after change

```html
1. Proposed  html of the page with the language attribute added 
<html lang="en" xmlns="http://www.w3.org/1999/xhtml">
<head>

2. Proposed html of Images having a title attribute value but no alt value.

<a href="books.html"><img
title="Packt Publishing donates a portion of the proceeds from this book to
support IPython's development." alt="How proceeds from this book are used" width="160" height="200" id="sidebar-book-cover"
style="margin-bottom: 10px;"></a>

3.Proposed html of PDF documents on the page (under documentation for older releases)
Proposed html of the pdf documents
<li>0.11: <a class="reference external" href="https://ipython.org/ipython-doc/rel-0.11/index.html">HTML</a> and <a title="open pdf file about version 0.11" class="reference external" href="https://ipython.org/ipython-doc/rel-0.11/ipython.pdf">PDF</a>.</li>
<li>0.10.2: <a class="reference external" href="https://ipython.org/ipython-doc/rel-0.10.2/html">HTML</a> and <a title= "open pdf file about version 0.10.2"class="reference external" href="https://ipython.org/ipython-doc/rel-0.10.2/ipython.pdf">PDF</a>.</li>
<li>0.10.1: <a class="reference external" href="https://ipython.org/ipython-doc/rel-0.10.1/html">HTML</a> and <a title="open pdf file about version 0.10.1"class="reference external" href="https://ipython.org/ipython-doc/rel-0.10.1/ipython.pdf">PDF</a>.</li>
<li>0.10: <a class="reference external" href="https://ipython.org/ipython-doc/rel-0.10/html">HTML</a> and <a title="open pdf file about version 0.10"class="reference external" href="https://ipython.org/ipython-doc/rel-0.10/ipython.pdf">PDF</a>.</li>
<li>0.9.1: <a class="reference external" href="https://ipython.org/ipython-doc/rel-0.9.1/html">HTML</a> and <a title ="open pdf file about version 0.9.1"class="reference external" href="https://ipython.org/ipython-doc/rel-0.9.1/ipython.pdf">PDF</a>.</li>
<li>0.9: <a class="reference external" href="https://ipython.org/ipython-doc/rel-0.9/html">HTML</a> and <a title="open pdf file about version 0.9"class="reference external" href="https://ipython.org/ipython-doc/rel-0.9/ipython.pdf">PDF</a>.</li>

4. Suggested html for the first image
<img class="logo" src="_static/IPy_header.png" alt="Logo of ipython interactive computing "/>

Suggested html of the second image (added a proper alt text and closed the image tag
<img src="_static/jupytercon-logo.svg" width="180px" alt="JupyterCon logo"/>

5. Css of the proposed footer 
*{
    margin: 0;
    padding: 0;
    font-family:'Poppins', 'sans-serif';
    box-sizing: border-box;
}
.footer {
margin-top: 150px;
width: 100%;
padding: 100px 15%;
background: #333;
color: #efefef;
display: flex;

}
.footer div{
    text-align: center;

}
.col-2{
    flex-grow: 2;
}
.footer div h3{
    font-weight: 300;
    margin-bottom: 30px;
    letter-spacing:1px ; 
}
.col-1 a{
display: block;
text-decoration: none;
color: #efefef;
margin-bottom: 10px;
}
form input{
    width: 400px;
    height: 45px;
    border-radius: 4px;
    text-align:center;
    margin-top: 20px;
    margin-bottom: 40px;
    outline: none;
    border: none;
}
form button{
    background: transparent;
    border: 2px solid #fff;
    color: #fff;
    border-radius: 30px;
    padding: 10px 30px;
    font-size: 15px;
    cursor: pointer;
}
.social-icons{
margin-top: 30px;

}
.social-icons i {
    font-size: 22px;
    margin: 10px;
    cursor: pointer;
}
.footer-bottom{
    text-align: center;
}

HTML of the proposed footer 
!DOCTYPE <!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>My Footer Design</title>
    <link rel="stylesheet" href="style.css">
    <script src="https://kit.fontawesome.com/fde0113670.js" crossorigin="anonymous"></script>
</head>
<body>
    <div class="footer">
<div class="col-1">
    <h3>USEFUL LINKS</h3>
    <a href="#">About Us</a>
    <a href="#">Services</a>
    <a href="#">Contact</a>
    <a href="#">Blog</a>
    <a href="#">Shop</a>
</div>
<div class="col-2">
<h3>NEWS LETTER</h3>
<form>
    <input type = "email" placeholder="Your Email Address"required>
    <br>
    <button type="submit">SUBSCRIBE NOW</button>
</form>
</div>
<div class="col-3">
<h3>GET IN TOUCH</h3>
<p> 123, XYZ Road </p>
<div class="social-icons">
    <i class="fa-brands fa-facebook"></i>
    <i class="fa-brands fa-twitter"></i>
    <i class="fa-brands fa-instagram"></i>
    <i class="fa-brands fa-linkedin"></i>
 </div>
</div>

</div>
<div class="footer-bottom">
    <div class="" role="contentinfo">
        &#169; Copyright the IPython development team.
      
    </div>
  
 </div> 
</body>
</html>

6. Proposed html to remove the aria label (also with the role attribute removed)
 <div class="sphinxsidebar">

7. Proposed html wi<script type="text/javascript" id="documentation_options"  src="_static/documentation_options.js"></script>th attribute "data-url_root" removed.

 8.Proposed html with the required attribute "type" specified
<script type="text/javascript" src="_static/nbviewer_thm.js"></script>
```

### More info

1.  Some headings are not structured well in the page

Suggestion: This can be improved by ensuring that where headings are used, the text in question is actually a heading and that it is structured correctly in the page outline. When this is put into consideration, the headings will facilitate page navigation for users of assistive technologies. The headings will also provide semantic and visual meaning and structure to the document.

2. Un Ordered lists were used
Suggestion: Where list items are sequential or numbered, an ordered list is likely more appropriate.
We should use unordered (bulleted) lists if they are appropriate for the context.

3. Ensuring that interactive elements like links and buttons are easy to identify while making sure that the naming of these elements is consistent throughout the website.

4. Some people can't use a mouse and use only a keyboard to navigate through web pages. 

It is important that users can reach all interactive elements using the keyboard, and that it is clear which element has focus.

Visible keyboard focus could be a border or highlight that moves as you tab through the web page.

**Comment by sgibson91** [comment]
Thank you for your contribution! Since the contribution period is now over, we will close this issue.
